app/controllers/identities_controller.rb

```diff
@@ -69,4 +80,29 @@ def show_sites # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity,
       render plain: "404 Not Found", status: :not_found
     end
   end
+
+  private
+
+  def flipper_params
+    params.require(:portal).to_sym
+  end
+
+  def form_user
+    identity = current_user.identity_scope.find_by(uid: params[:id])
+    identity.user.becomes(FormUser)
+  end
+
+  def admin?
+    current_user.user_actions&.dig("admin", "can_administer")
+  end
+
+  def switch_portal(portal, form_user)
+    if portal.enabled?(form_user)
+      portal.disable(form_user)
+      flash[:error] = "Portal successfully disabled"
```

Comment:
is this an error?

Reply:
put for having different colour notification during development, will return to `success`. cheers!